druid/src/command.rs

```diff
@@ -73,19 +122,42 @@ enum Arg {
     OneShot(Arc<Mutex<Option<Box<dyn Any>>>>),
 }
 
-/// Errors that can occur when attempting to retrieve the a command's argument.
+/// Errors that can occur when attempting to retrieve the `OneShotCommand`s argument.
 #[derive(Debug, Clone, PartialEq)]
 pub enum ArgumentError {
-    /// The command did not have an argument.
-    NoArgument,
-    /// The argument was expected to be reusable and wasn't, or vice-versa.
-    WrongVariant,
-    /// The argument could not be downcast to the specified type.
-    IncorrectType,
+    /// The command represented a different selector.
+    WrongSelector,
     /// The one-shot argument has already been taken.
     Consumed,
 }
 
+/// This error can occur when wrongly promising that a type erased
+/// variant of some generic item represents the application state.
+/// Examples are `MenuDesc<T>` and `AppStateMenuDesc`.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct AppStateTypeError {
```

Comment:
do we have an 'errors' module? I might move that there, I'm not sure.

I'd also maybe be more general, and call it a `DowncastError` or something?

Reply:
I don't think we have one, but it could be interesting to see if it would be beneficial.
As for the name, `DowncastError` is less specific but would be more in line with the other names.